### PR TITLE
NSB v9 upgrade and cleanup

### PR DESCRIPTION
**Pull request by danielmarbach**
FYI This PR started as an attempt to make the demo cross platform, but to make this happen, we'll need to add some way to have an inter-process communication channel.

Instead, we've cleaned up the repo and upgraded everything to NServiceBus 9.

### CODE REVIEW

**Review by mauroservienti — DISMISSED** [review]

**Comment by danielmarbach** [comment]
@bording @mauroservienti  maybe we can discuss how to proceed here together. I went down the path of single file publishing but stopped midway because I figured we might want to align first

**Comment by danielmarbach** [comment]
Maybe a pragmatic intermediate step is to remove the single file ms build property for now since we can always do that later

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -12,7 +12,7 @@ jobs:
       - name: Setup .NET SDK
         uses: actions/setup-dotnet@v4.0.0
         with:
-          dotnet-version: 5.0.x
+          dotnet-version: 8.0.x
```
Comment (danielmarbach) [inline .github/workflows/release.yml:15]: Signing part requires path updates

File: src/Billing/Billing.csproj [hunk 2]
```diff
@@ -1,19 +1,23 @@
-<?xml version="1.0" encoding="utf-8"?>
 <Project Sdk="Microsoft.NET.Sdk">
+
   <PropertyGroup>
-    <TargetFramework>net472</TargetFramework>
+    <TargetFramework>net8.0</TargetFramework>
     <OutputType>Exe</OutputType>
-    <LangVersion>7.1</LangVersion>
+    <PublishSingleFile>true</PublishSingleFile>
```
Comment (danielmarbach) [inline src/Billing/Billing.csproj:6]: Probably remove for now

**Comment by danielmarbach** [comment]
Removed the single file publish part for now

**Review by mauroservienti — DISMISSED** [review]

**Review by bording — DISMISSED** [review]

**Review by bording — APPROVED** [review]

**Pull request merged**